include/boost/geometry/algorithms/detail/envelope/segment.hpp

```diff
@@ -334,7 +333,7 @@ class envelope_segment_impl
         CalculationType lon2_rad = math::as_radian<Units>(lon2);
         CalculationType lat2_rad = math::as_radian<Units>(lat2);
         CalculationType alp1, alp2;
-        strategy.apply(lon1_rad, lat1_rad, lon2_rad, lat2_rad, alp1, alp2);
+        strategy.apply<true, true>(lon1_rad, lat1_rad, lon2_rad, lat2_rad, alp1, alp2);
```

Comment:
I prefer the first proposed interface (which I didn't implement before because I wanted to stick on single "apply" name but I agree it is cleaner). The second interface above call two times the azimuth formula when both alph1 and alph2 needed. Regarding switched endpoints it is not always true and that is the reason for this change here. Sometimes you have to subtract/add pi etc but this is formula's job.

Reply:
The second interface only presents the two latter functions that are different. The one taking 2 output parameters would also be there. So no double calculation also in this case.